code/game/objects/items/devices/scanners.dm

```diff
@@ -482,11 +482,8 @@ Subject's pulse: ??? BPM"})
 	var/details = 0
 	var/recent_fail = 0
 
-/obj/item/device/reagent_scanner/afterattack(obj/O, mob/user as mob)
-	. = ..()
-	if(.)
-		return
-	if(!istype(O)) //Wrong type sent
+/obj/item/device/reagent_scanner/preattack(atom/O, mob/user)
+	if(!user.Adjacent(O))
```

Comment:
use the `proximity_flag` var instead
preattack is defined as `/obj/item/proc/preattack(atom/target, mob/user, proximity_flag, click_parameters)`